launcher/archive/ArchiveWriter.cpp

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2025 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "ArchiveWriter.h"
+#include <archive.h>
+#include <archive_entry.h>
+#include <sys/stat.h>
+
+#include <QFile>
+#include <QFileInfo>
+
+#include <memory>
+
+namespace MMCZip {
+
+ArchiveWriter::ArchiveWriter(const QString& archiveName) : m_filename(archiveName) {}
+
+ArchiveWriter::~ArchiveWriter()
+{
+    close();
+}
+
+bool ArchiveWriter::open()
+{
+    if (m_filename.isEmpty()) {
+        qCritical() << "Archive m_filename not set.";
+        return false;
+    }
+
+    m_archive = archive_write_new();
+    if (!m_archive) {
+        qCritical() << "Archive not initialized.";
+        return false;
+    }
+
+    archive_write_set_format_zip(m_archive);
```

Comment:
```suggestion
    archive_write_set_format_by_name(m_archive, m_format);
```
Small change to make writing other formats actually possible